leader-election-impl/src/test/java/com/palantir/paxos/VerifyingPaxosStateLogTest.java

```diff
@@ -0,0 +1,111 @@
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class VerifyingPaxosStateLogTest {
+    private static final long LEAST_LOG_ENTRY = 3L;
+    private static final long GREATEST_LOG_ENTRY = 20L;
+    private static final byte[] CORRECT_VALUE_FOR_ROUND = valueForRound(10L).persistToBytes();
+    private static final byte[] OTHER_VALUE_FOR_ROUND = valueForRound(5L).persistToBytes();
+
+    private final PaxosStateLog<PaxosValue> current = mock(PaxosStateLog.class);
+    private final PaxosStateLog<PaxosValue> experimental = mock(PaxosStateLog.class);
+    private final VerifyingPaxosStateLog.Settings<PaxosValue> settings = ImmutableSettings.<PaxosValue>builder()
+            .currentLog(current)
+            .experimentalLog(experimental)
+            .hydrator(PaxosValue.BYTES_HYDRATOR)
+            .build();
+
+    private final PaxosStateLog<PaxosValue> log = new VerifyingPaxosStateLog<>(settings);
+
+    @Before
+    public void setupMocks() throws IOException {
+        when(current.getLeastLogEntry()).thenReturn(LEAST_LOG_ENTRY);
+        when(current.getGreatestLogEntry()).thenReturn(GREATEST_LOG_ENTRY);
+        when(current.readRound(anyLong())).thenReturn(CORRECT_VALUE_FOR_ROUND);
+
+        when(experimental.getLeastLogEntry()).thenReturn(LEAST_LOG_ENTRY + 1L);
+        when(experimental.getGreatestLogEntry()).thenReturn(GREATEST_LOG_ENTRY + 1L);
+        when(experimental.readRound(anyLong())).thenReturn(OTHER_VALUE_FOR_ROUND);
+    }
+
+    @After
+    public void noMoreInteractions() {
+        verifyNoMoreInteractions(current, experimental);
+    }
+
+    @Test
+    public void writeWritesToBoth() {
+        long sequence = 15L;
+        log.writeRound(sequence, valueForRound(sequence));
+        verify(current).writeRound(sequence, valueForRound(sequence));
+        verify(experimental).writeRound(sequence, valueForRound(sequence));
+    }
+
+    @Test
+    public void readFromBothReturnLegacy() throws IOException {
+        long sequence = 10L;
+        assertThat(log.readRound(sequence)).isEqualTo(CORRECT_VALUE_FOR_ROUND);
+        verify(current).readRound(sequence);
+        verify(experimental).readRound(sequence);
+    }
+
+    @Test
+    public void getLeastFromBothReturnLegacy() {
+        assertThat(log.getLeastLogEntry()).isEqualTo(LEAST_LOG_ENTRY);
+        verify(current).getLeastLogEntry();
+        verify(experimental).getLeastLogEntry();
+    }
+
+    @Test
+    public void getGreatestFromBothReturnLegacy() {
+        assertThat(log.getGreatestLogEntry()).isEqualTo(GREATEST_LOG_ENTRY);
+        verify(current).getGreatestLogEntry();
+        verify(experimental).getGreatestLogEntry();
+    }
+
+    @Test
+    public void truncateBoth() {
+        log.truncate(7L);
+        verify(current).truncate(7L);
+        verify(experimental).truncate(7L);
+    }
+
```

Comment:
Might want to add a few tests for the exception protection for experimental, once we've implemented that!